vello/src/scene.rs

```diff
@@ -333,24 +344,340 @@ impl<'a> DrawGlyphs<'a> {
+            Point::new(x.into(), y.into()),
+        );
+    }
+
+    fn curve_to(&mut self, cx0: f32, cy0: f32, cx1: f32, cy1: f32, x: f32, y: f32) {
+        self.0.curve_to(
+            Point::new(cx0.into(), cy0.into()),
+            Point::new(cx1.into(), cy1.into()),
+            Point::new(x.into(), y.into()),
+        );
+    }
+
+    fn close(&mut self) {
+        self.0.close_path();
+    }
+}
+
+impl DrawColorGlyphs<'_> {
+    fn last_transform(&self) -> Transform {
+        self.transform_stack
+            .last()
+            .copied()
+            .unwrap_or(Transform::IDENTITY)
+    }
+}
+
+fn conv_skrifa_transform(transform: skrifa::color::Transform) -> Transform {
+    Transform {
+        matrix: [transform.xx, transform.xy, transform.yx, transform.yy],
+        translation: [transform.dx, transform.dy],
+    }
+}
+
+fn conv_brush(brush: skrifa::color::Brush, cpal: &Cpal<'_>) -> Brush {
+    match brush {
+        skrifa::color::Brush::Solid {
+            palette_index,
+            alpha,
+        } => Brush::Solid(color_index(cpal, palette_index).with_alpha_factor(alpha)),
+        skrifa::color::Brush::LinearGradient {
+            p0,
+            p1,
+            color_stops,
+            extend,
+        } => Brush::Gradient(
+            Gradient::new_linear(conv_point(p0), conv_point(p1))
+                .with_extend(conv_extend(extend))
+                .with_stops(ColorStopsConverter(color_stops, cpal)),
+        ),
+        skrifa::color::Brush::RadialGradient {
+            c0,
+            r0,
+            c1,
+            r1,
+            color_stops,
+            extend,
+        } => Brush::Gradient(
+            Gradient::new_two_point_radial(conv_point(c0), r0, conv_point(c1), r1)
+                .with_extend(conv_extend(extend))
+                .with_stops(ColorStopsConverter(color_stops, cpal)),
+        ),
+        skrifa::color::Brush::SweepGradient {
+            c0,
+            start_angle,
+            end_angle,
+            color_stops,
+            extend,
+        } => Brush::Gradient(
+            Gradient::new_sweep(conv_point(c0), start_angle, end_angle)
+                .with_extend(conv_extend(extend))
+                .with_stops(ColorStopsConverter(color_stops, cpal)),
+        ),
+    }
+}
+
+fn color_index(cpal: &'_ Cpal<'_>, palette_index: u16) -> Color {
+    let actual_colors = cpal.color_records_array().unwrap().unwrap();
+    let color = actual_colors[usize::from(palette_index)];
+    Color::rgba8(color.red, color.green, color.blue, color.alpha)
+}
+
+fn conv_point(point: skrifa::raw::types::Point<f32>) -> Point {
+    Point::new(point.x.into(), point.y.into())
+}
+
+fn conv_extend(extend: skrifa::color::Extend) -> Extend {
+    match extend {
+        skrifa::color::Extend::Pad => Extend::Pad,
+        skrifa::color::Extend::Repeat => Extend::Reflect,
+        skrifa::color::Extend::Reflect => Extend::Repeat,
+        // TODO: Error reporting on unknown variant?
```

Comment:
This seems like a reasonable default. Lots of broken fonts in the wild and there’s not much we can do about it now :(